### PR TITLE
Finalise

### PR DESCRIPTION
**Pull request by FranzGithub**
This PR introduces a bunch of fixes.

### CODE REVIEW

File: src/main/java/se/lth/base/server/data/User.java [hunk 1]
```diff
@@ -75,6 +78,14 @@ public String getFirstName() {
   public double getRating() {
     return rating;
   }
+  /**
+   * Gets the rating of the user
```
Comment (jadlers) [inline src/main/java/se/lth/base/server/data/User.java]: Should be sets not gets, right?

```suggestion
   * Sets the rating of the user
```

File: src/main/resources/webassets/passenger/passengerSearch.js [hunk 2]
```diff
@@ -11,6 +28,7 @@ function joinTrip(btn) {
 }
 
 base.passengerSearchController = function() {
+  if (!map) initMap();
```
Comment (jadlers) [inline src/main/resources/webassets/passenger/passengerSearch.js]: Move this to the `load` function in the controller. Should not load until needed (and will most likely break my future changes if left here)

File: src/main/resources/webassets/rest.js [hunk 3]
```diff
@@ -122,6 +125,20 @@ base.rest = (function() {
...
     },
+    addRating: function(target_id, rating) {
+      return baseFetch('/rest/user/' + target_id + '/addrating/' + rating, {
+        method: 'POST',
+        body: JSON.stringify(credentials),
+        headers: jsonHeader,
+      }).then(response => response.json());
+    },
+    setRating: function(id, rating) {
+      return baseFetch('/rest/user/' + target_id + '/setrating/' + rating, {
```
Comment (jadlers) [inline src/main/resources/webassets/rest.js]: `target_id` is not defined here. Should it be `id`?
```suggestion
      return baseFetch('/rest/user/' + id + '/setrating/' + rating, {
```

File: src/main/java/se/lth/base/server/rest/UserResource.java [hunk 4]
```diff
@@ -109,19 +109,20 @@ public User getUser(@PathParam("id") int userId) {
...
   }
 
-  @Path("{id}")
+  @Path("{id}/{rating}")
   @RolesAllowed(Role.Names.ADMIN)
   @PUT
   @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
   @Consumes(MediaType.APPLICATION_JSON + ";charset=utf-8")
-  public User putUser(@PathParam("id") int userId, Credentials credentials) {
+  public User putUser(
```
Comment (jadlers) [inline src/main/java/se/lth/base/server/rest/UserResource.java]: Should this change the purpose of `putUser` I'm not sure if it's used but maybe a new function named something like `setRating` would be better?

File: src/main/resources/webassets/admin/user-admin.html [hunk 5]
```diff
@@ -50,6 +50,12 @@ <h1>User Admin</h1>
                     </div>
                 </div>
             </div>
+            <div class="form-group row show-edit">
+                <label class="col-md-2 col-form-label" for="set-rating">Rating</label>
+                <div class="col-md-10">
+                    <input type="text" class="form-control" id="set-rating" pattern="[12345]{1}" required>
```
Comment (jadlers) [inline src/main/resources/webassets/admin/user-admin.html]: Should be a type="number"?

File: src/main/resources/webassets/admin/user-admin.js [hunk 6]
```diff
@@ -82,14 +83,15 @@ base.userAdminController = function() {
...
+      var rating = document.getElementById('set-rating').value;
       var role = document.getElementById('set-role').value;
       var id = document.getElementById('user-id').value;
       var credentials = { username, password, role };
       if (password === '') {
         delete credentials.password;
       }
       if (id !== '') {
-        base.rest.putUser(id, credentials).then(function(user) {
+        base.rest.putUser(id, rating, credentials).then(function(user) {
```
Comment (jadlers) [inline src/main/resources/webassets/admin/user-admin.js]: Same comment here as above, maybe a `setRating` function would be better?

File: src/main/java/se/lth/base/server/rest/UserResource.java (hunk 4 above)
Comment (FranzGithub) [reply src/main/java/se/lth/base/server/rest/UserResource.java]: Would be better! Given more time I would suggest that solution :p

File: src/main/resources/webassets/admin/user-admin.html (hunk 5 above)
Comment (FranzGithub) [reply src/main/resources/webassets/admin/user-admin.html]: number cannot have pattern and will thus allow e

File: src/main/resources/webassets/admin/user-admin.js (hunk 6 above)
Comment (FranzGithub) [reply src/main/resources/webassets/admin/user-admin.js]: True! Pls do not make me do this!

**Pull request merged**